SCons/Util.py

```diff
@@ -2123,6 +2124,40 @@ def print_time():
     from SCons.Script.Main import print_time
     return print_time
 
+
+def wait_for_process_to_die(pid):
+    """
+    Wait for specified process to die, or alternatively kill it
+    NOTE: This function operates best with psutil pypi package
+    """
+    # wait for the process to fully killed
+    try:
+        import psutil
+        while True:
```

Comment:
Just want to note this loop could be troublesome if for some reason the process wont die. Could be a default time out and related error message to save future headaches

Reply:
@dmoody256 - what seems like a reasonable timeout? probably as a parameter to this.. maybe 5 seconds?